config/src/iroha.rs

```diff
@@ -151,79 +148,101 @@ impl ConfigurationProxy {
     ///
     /// # Errors
     /// - Finalisation fails
-    /// - Any of the inner fields had a `None` value when that
+    /// - Building fails, e.g. any of the inner fields had a `None` value when that
     /// is not allowed by the defaults.
     pub fn build(mut self) -> Result<Configuration> {
-        self.finalize()?;
-        <Self as Builder>::build(self)
+        self.finish()?;
+        <Self as iroha_config_base::proxy::Builder>::build(self)
             .wrap_err("Failed to build `Configuration` from `ConfigurationProxy`")
     }
 }
 
-impl Configuration {
-    /// Construct [`struct@Self`] from a path-like object.
-    ///
-    /// # Errors
-    /// - File not found.
-    /// - File found, but peer configuration parsing failed.
-    /// - The length of the array in raw JSON representation is different
-    /// from the length of the array in
-    /// [`self.sumeragi.trusted_peers.peers`], most likely due to two
-    /// (or more) peers having the same public key.
-    pub fn from_path<P: AsRef<Path> + Debug + Clone>(path: P) -> Result<Configuration> {
-        let file = File::open(path.clone())
-            .wrap_err(format!("Failed to open the config file {:?}", path))?;
-        let reader = BufReader::new(file);
-        let mut configuration: Configuration = serde_json::from_reader(reader).wrap_err(
-            format!("Failed to parse {:?} as Iroha peer configuration.", path),
-        )?;
-        configuration.finalize()?;
-        Ok(configuration)
-    }
-
-    fn finalize(&mut self) -> Result<()> {
-        self.sumeragi.key_pair = KeyPair::new(self.public_key.clone(), self.private_key.clone())?;
-        self.sumeragi.peer_id =
-            iroha_data_model::peer::Id::new(&self.torii.p2p_addr, &self.public_key.clone());
-
-        Ok(())
-    }
-
-    /// Load configuration from the environment
-    ///
-    /// # Errors
-    /// Fails if Configuration deserialization fails (e.g. if `TrustedPeers` contains entries with duplicate public keys)
-    pub fn load_environment(&mut self) -> Result<()> {
-        <Self as iroha_config_base::proxy::LoadFromEnv>::load_environment(self)?;
-        self.finalize()?;
-        Ok(())
-    }
-}
-
 #[cfg(test)]
 mod tests {
     #![allow(clippy::restriction)]
 
+    use proptest::prelude::*;
+
     use super::*;
     use crate::{base::proxy::LoadFromDisk, sumeragi::TrustedPeers};
 
-    const CONFIGURATION_PATH: &str = "../configs/peer/config.json";
+    const CONFIGURATION_PATH: &str = "../configs/peer/config_test.json";
+
+    /// Key-pair used for proptests generation
+    #[allow(clippy::expect_used)]
+    pub fn placeholder_keypair() -> KeyPair {
```

Comment:
Why not generate keys here as well?

Reply:
This is going to add stochasticity to the (already flaky) tests. We need to remove as many variables as we can, unless they are part of the proptest. In that case, the keys will need to be generated from a seed that's provide by the proptest framework.